### ISSUE TITLE
Max ttl of 20 years will overflow localDeletionTime

### ISSUE THREAD

**Issue by amoskong** [scylladb/scylla#3182]
*Installation details*
Scylla version (or git commit hash):  666.development-0.20180205.6919c7434
Cluster size: 1
OS (RHEL/CentOS/Ubuntu/AWS AMI): CentOS

Reference: [CASSANDRA-14092](https://issues.apache.org/jira/browse/CASSANDRA-14092)

Current code has a max limit (630720000, 20 years in second) for ttl value:
```
cqlsh> INSERT INTO keyspace1.standard1 (key,"C0") VALUES (textAsBlob('1'),
textAsBlob('Q3DRB7BJAMa')) USING TTL 2147483648;
InvalidRequest: Error from server: code=3D2200 [Invalid query]
message=3D"marshaling error: Value out of range for type
org.apache.cassandra.db.marshal.*Int32Type*: '2147483648'"

cqlsh> INSERT INTO keyspace1.standard1 (key,"C0") VALUES (textAsBlob('1'),
textAsBlob('Q3DRB7BJAMa')) USING TTL 2147483647;
InvalidRequest: Error from server: code=3D2200 [Invalid query] message=3D"t=
tl is too large. requested (2147483647) maximum (`630720000`)"
```

But the 630720000 isn't valid after `Wed Jan 24 11:14:07 2018`, current time + 20 years will overflow 2147483647.
```
>>> time.ctime((2**31 -1) - (20 * 365 * 3600 * 24))
'Wed Jan 24 11:14:07 2018'
```

Tests:
```
cqlsh> INSERT INTO keyspace1.standard1 (key,"C0") VALUES (textAsBlob('1'),
textAsBlob('3')) USING TTL 630720000;
cqlsh> select * from keyspace1.standard1 where key =3D textAsBlob('1');

 key | C0 | C1 | C2 | C3 | C4
-----+----+----+----+----+----

(0 rows)
```
Result: no error raised, but data isn't successfully inserted to table.

@tgrabiec

**Comment by slivne** [comment]
TTL is set as an offset not as an absolute value

Not clear to me whats the issue

**Comment by dorlaor** [comment]
The issue as reported on C* is that it wraps around if you set the TTL
above 20 years
and thus a TTL of 21 years becomes 1 year and here goes your data

**Comment by slivne** [comment]
If we are referencing a cassandra issue lets add a link to that.

**Comment by dorlaor** [comment]
It was noted (and also discussed on their mailing list).

Reference: CASSANDRA-14092

**Comment by avikivity** [comment]
@bhalevy I think you fixed it. Can you find the commit?

**Comment by bhalevy** [comment]
This is a duplicate of #3353 that was originally fixed with 93270dd8e04e3caf0e7586311badbaf121c14d3f
and then required 737382547318520772570bd6fb991782f74ea5b4

**Comment by avikivity** [comment]
Fixed by 737382547318520772570bd6fb991782f74ea5b4
